storey/sql_driver.py

```diff
@@ -87,17 +87,23 @@ async def close(self):
             self._sql_connection = None
 
     async def _get_all_fields(self, key, table):
-        where_clause = self._get_where_clause(key, table)
-        query = f"SELECT * FROM {table} where {where_clause}"
-        results = pd.read_sql(query, con=self._sql_connection, parse_dates=self._time_fields).to_dict(orient="records")
+        key = self._extract_list_of_keys(key)
+        select_object = db.select(table).where(
+            db.and_(getattr(table.c, self._primary_key[i]) == key[i] for i in range(len(self._primary_key)))
+        )
+        results = pd.read_sql(select_object, con=self._sql_connection, parse_dates=self._time_fields).to_dict(
```

Comment:
So you went with full DSL – cool. We need to do this in `SQLSource` too. It still uses string substitution. For the tests I suppose it doesn't matter that much, but still nice to have if it's not too much work.

Reply:
Speaking of tests, it would be great if we could add a test that tries to do SQL injection.